src/main/java/org/sonarlint/intellij/config/global/SonarQubeServerMgmtPanel.java

```diff
@@ -0,0 +1,258 @@
+      protected void customizeCellRenderer(JList list, Object value, int index, boolean selected, boolean hasFocus) {
+        SonarQubeServer server = (SonarQubeServer)value;
+        try {
+          setIcon(ResourceLoader.getIcon(LIST_ICON));
+        } catch (IOException e) {
+          e.printStackTrace();
+        }
+
+        String url;
+        if(server.getName() == null || server.getName().isEmpty()) {
+          url = LABEL_UNDEFINED;
+        } else {
+          url = server.getName();
+        }
+        append(url, SimpleTextAttributes.REGULAR_ATTRIBUTES);
+      }
+    });
+
+    changeListener = new Consumer<SonarQubeServer>() {
+      public void consume(SonarQubeServer server) {
+        ((CollectionListModel)serverList.getModel()).contentsChanged(server);
+      }
+    };
+  }
+
+  public JComponent getComponent() {
+    return panel;
+  }
+
+  public boolean isModified(SonarLintGlobalSettings settings) {
+    return !servers.equals(settings.getServers());
+  }
+
+  public void save(SonarLintGlobalSettings settings) {
+    List<SonarQubeServer> copyList = new LinkedList<>();
+    for(SonarQubeServer s : servers) {
+      copyList.add(new SonarQubeServer(s));
+    }
+    settings.setSonarQubeServers(copyList);
+  }
+
+  public void load(SonarLintGlobalSettings settings) {
+    serverIds.clear();
+    editorPanel.removeAll();
+    editorPanel.add(emptyPanel, EMPTY_CARD_NAME);
+    servers.clear();
+
+    CollectionListModel<SonarQubeServer> listModel = new CollectionListModel<>(new ArrayList<SonarQubeServer>());
+    for (SonarQubeServer s : settings.getServers()) {
+      SonarQubeServer copy = new SonarQubeServer(s);
+      listModel.add(copy);
+      servers.add(copy);
+      addServerEditor(copy);
+    }
+
+    serverList.setModel(listModel);
+
+    if (!servers.isEmpty()) {
+      serverList.setSelectedValue(servers.get(0), true);
+    }
+  }
+
+  @Nullable
+  private SonarQubeServer getSelectedServer() {
+    return (SonarQubeServer) serverList.getSelectedValue();
+  }
+
+  private void addServerEditor(SonarQubeServer server) {
+    SonarQubeServerEditorPanel editor = new SonarQubeServerEditorPanel(changeListener, server);
+
+    JComponent component = editor.create();
+    String name = serverIds.get(server);
+    editorPanel.add(component, name);
+    editorPanel.doLayout();
+  }
+
+  private void addServer(SonarQubeServer server) {
+    servers.add(server);
+    ((CollectionListModel)serverList.getModel()).add(server);
+    addServerEditor(server);
+    serverList.setSelectedIndex(serverList.getModel().getSize() - 1);
+  }
+
+  private class AddServerAction extends IconWithTextAction implements DumbAware {
+    public AddServerAction() {
+      super("SonarQube server", "New SonarQube server", null);
+      setIcon();
+    }
+
+    private void setIcon() {
+      try {
+        Icon icon = ResourceLoader.getIcon(LIST_ICON);
+        super.getTemplatePresentation().setIcon(icon);
+      } catch (IOException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1166)